From 4595453e4ff31019ccaa9732bbef5106f6d46e6b Mon Sep 17 00:00:00 2001
From: Fam Zheng <famz@redhat.com>
Date: Tue, 4 Nov 2014 03:06:40 +0100
Subject: [PATCH 6/9] virtio-scsi: Fix memory leak when realize failed

Message-id: <1415070401-21222-3-git-send-email-famz@redhat.com>
Patchwork-id: 62061
O-Subject: [RHEL-7.1 qemu-kvm PATCH v2 2/3] virtio-scsi: Fix memory leak when realize failed
Bugzilla: 1089606
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Max Reitz <mreitz@redhat.com>

Signed-off-by: Fam Zheng <famz@redhat.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 93bd49aff9081bbe9440192db9da3676941f77a3)
Signed-off-by: Fam Zheng <famz@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>

Conflicts:
	hw/scsi/virtio-scsi.c
Context conflict due to previous patch's resolution.
---
 hw/scsi/virtio-scsi.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hw/scsi/virtio-scsi.c b/hw/scsi/virtio-scsi.c
index 73f5ad1..13df89d 100644
--- a/hw/scsi/virtio-scsi.c
+++ b/hw/scsi/virtio-scsi.c
@@ -609,6 +609,7 @@ int virtio_scsi_common_init(VirtIOSCSICommon *s)
         error_report("Invalid number of queues (= %" PRId32 "), "
                          "must be a positive integer less than %d.",
                      s->conf.num_queues, VIRTIO_PCI_QUEUE_MAX);
+        virtio_cleanup(vdev);
         return -1;
     }
     s->cmd_vqs = g_malloc0(s->conf.num_queues * sizeof(VirtQueue *));
-- 
1.8.3.1

